rust/actyx/ax/tests/cli.rs

```diff
@@ -109,82 +107,32 @@ fn cli_fail_on_missing_identity() {
 
 #[test]
 fn internal_subcommand() {
-    cli().args(["internal", "help"]).assert().failure();
+    use predicate::str::contains;
+    use predicates::prelude::*;
     cli()
         .env("HERE_BE_DRAGONS", "wrong")
-        .args(["internal", "help"])
         .assert()
-        .failure();
+        .failure()
+        .stderr(contains("internal").not());
+    cli()
+        .args(["internal"])
+        .assert()
+        .failure()
+        .stderr(contains("do not use until instructed by Actyx"));
     cli()
         .env("HERE_BE_DRAGONS", "zøg")
-        .args(["internal", "help"])
         .assert()
-        .success();
+        .failure()
+        .stderr(contains("internal"));
     cli()
         .env("HERE_BE_DRAGONS", "zoeg")
-        .args(["internal", "help"])
         .assert()
-        .success();
+        .failure()
+        .stderr(contains("internal"));
 }
 
 #[test]
 fn version() {
     let first_line = format!("ax {}\n", NodeVersion::get());
     cli().arg("--version").assert().stdout(first_line).success();
-
-    #[derive(PartialEq)]
-    enum Type {
-        Branch,
-        Leaf,
-    }
-    use predicate::str::starts_with;
-    use std::iter::once;
-    use Type::*;
-
-    let commands = btreemap! {
-        vec!["apps"] => Branch,
-        vec!["apps", "sign"] => Leaf,
-        vec!["events"] => Branch,
-        vec!["events", "offsets"] => Leaf,
-        vec!["events", "query"] => Leaf,
-        vec!["events", "publish"] => Leaf,
-        vec!["events", "dump"] => Leaf,
-        vec!["events", "restore"] => Leaf,
-        vec!["internal"] => Branch,
-        vec!["internal", "trees"] => Branch,
-        vec!["internal", "trees", "dump"] => Leaf,
-        vec!["internal", "trees", "explore"] => Leaf,
-        vec!["nodes"] => Branch,
-        vec!["nodes", "inspect"] => Leaf,
-        vec!["nodes", "ls"] => Leaf,
-        vec!["settings"] => Branch,
-        vec!["settings", "schema"] => Leaf,
-        vec!["settings", "get"] => Leaf,
-        vec!["settings", "set"] => Leaf,
-        vec!["settings", "unset"] => Leaf,
-        vec!["swarms"] => Branch,
-        vec!["swarms", "keygen"] => Leaf,
-        vec!["users"] => Branch,
-        vec!["users", "keygen"] => Leaf,
-        vec!["users", "add-key"] => Leaf,
-    };
-
-    let first_line = |sub| format!("ax-{} {}\n", sub, NodeVersion::get());
-    for (args, tpe) in commands {
-        if tpe == Branch {
-            cli()
-                .args(&*args)
-                .env("HERE_BE_DRAGONS", "zøg")
-                .assert()
-                .failure()
-                .stderr(starts_with(&*first_line(args.join("-"))));
-        }
-        let name = args.join("-");
-        cli()
-            .args(&*args.into_iter().chain(once("--help")).collect::<Vec<_>>())
-            .env("HERE_BE_DRAGONS", "zøg")
-            .assert()
-            .success()
-            .stdout(starts_with(&*first_line(name)));
-    }
```

Comment:
Since we want to guarantee semver also for the command-line behaviour, I think we should keep this test (and update the list of sub-commands to test).

Reply:
My suggestion is to use `ax <subcommands>... --version`, assuming that that fails for non-existing subcommands.

Reply:
Ok, let me try a bit if that can be used for it. We can also compare them with default help if it makes sense.

Reply:
fixed. could you do a quick check @rkuhn?